### PR TITLE
#554 embed commands

### PR DESCRIPTION
**Pull request by ethanfrey**
Fix issue #554 to make it easier to maintain consistency between stand-alone tendermint, and tendermint embedded in basecoin.

This should make it into 0.10.1

### CODE REVIEW

**Comment by ethanfrey** [comment]
Works out with embedding for basecoin 0.6.1.

@ebuchman can you review and merge?

File: cmd/tendermint/commands/run_node.go [hunk 1]
```diff
@@ -19,28 +20,48 @@ var runNodeCmd = &cobra.Command{
...
+	cmd.Flags().String("rpc.laddr", config.RPC.ListenAddress, "RPC listen address. Port required")
+	cmd.Flags().String("rpc.grpc_laddr", config.RPC.GRPCListenAddress, "GRPC listen address (BroadcastTx only). Port required")
+	cmd.Flags().Bool("rpc.unsafe", config.RPC.Unsafe, "Enabled unsafe rpc methods")
 
 	// p2p flags
-	runNodeCmd.Flags().String("p2p.laddr", config.P2P.ListenAddress, "Node listen address. (0.0.0.0:0 means any interface, any port)")
-	runNodeCmd.Flags().String("p2p.seeds", config.P2P.Seeds, "Comma delimited host:port seed nodes")
-	runNodeCmd.Flags().Bool("p2p.skip_upnp", config.P2P.SkipUPNP, "Skip UPNP configuration")
-	runNodeCmd.Flags().Bool("p2p.pex", config.P2P.PexReactor, "Enable Peer-Exchange (dev feature)")
+	cmd.Flags().String("p2p.laddr", config.P2P.ListenAddress, "Node listen address. (0.0.0.0:0 means any interface, any port)")
```
Comment (ebuchman) [inline cmd/tendermint/commands/run_node.go]: Why doesn't the diff highlight the words that changed here like it does above ?

File: cmd/tendermint/commands/run_node.go [hunk 2]
```diff
@@ -19,28 +20,48 @@ var runNodeCmd = &cobra.Command{
...
+	cmd.Flags().String("p2p.seeds", config.P2P.Seeds, "Comma delimited host:port seed nodes")
+	cmd.Flags().Bool("p2p.skip_upnp", config.P2P.SkipUPNP, "Skip UPNP configuration")
+	cmd.Flags().Bool("p2p.pex", config.P2P.PexReactor, "Enable Peer-Exchange (dev feature)")
+}
 
-	RootCmd.AddCommand(runNodeCmd)
+func ParseGenesisFile(genDocFile string) (*types.GenesisDoc, error) {
```
Comment (ebuchman) [inline cmd/tendermint/commands/run_node.go]: Should we just put this in `types` ?

File: cmd/tendermint/commands/run_node.go [hunk 3]
```diff
@@ -56,27 +77,16 @@ func runNode(cmd *cobra.Command, args []string) error {
 	// TODO: If Mintnet gets deprecated or genesis_file is
 	// always available, remove.
 	genDocFile := config.GenesisFile()
-	if !cmn.FileExists(genDocFile) {
+	for !cmn.FileExists(genDocFile) {
```
Comment (ebuchman) [inline cmd/tendermint/commands/run_node.go]: I think we can get rid of this behaviour. If no genesis file, just fail.

File: types/genesis.go [hunk 4]
```diff
@@ -39,6 +39,15 @@ func (genDoc *GenesisDoc) SaveAs(file string) error {
 	return cmn.WriteFile(file, genDocBytes, 0644)
 }
 
+func (genDoc *GenesisDoc) ValidatorHash() []byte {
```
Comment (ebuchman) [inline types/genesis.go]: Awesome

Comment (ebuchman) [inline types/genesis.go]: Should we canonicalize the order tho ?

File: cmd/tendermint/commands/run_node.go (hunk 2 above)
Comment (ebuchman) [inline cmd/tendermint/commands/run_node.go]: This is `state.MakeGenesisStateFromFile` too

Comment (ethanfrey) [inline cmd/tendermint/commands/run_node.go]: ugh.  please clean up and let me know where the code ends up.

i don't know this repo very well.

File: cmd/tendermint/commands/run_node.go (hunk 3 above)
Comment (ethanfrey) [inline cmd/tendermint/commands/run_node.go]: what is mintnet?

File: types/genesis.go (hunk 4 above)
Comment (ethanfrey) [inline types/genesis.go]: NewValidatorSet does that :)

Comment (ebuchman) [inline types/genesis.go]: Oh good

File: cmd/tendermint/commands/run_node.go (hunk 3 above)
Comment (ebuchman) [inline cmd/tendermint/commands/run_node.go]: Deprecated tool https://github.com/tendermint/mintnet

File: cmd/tendermint/commands/run_node.go (hunk 1 above)
Comment (ethanfrey) [inline cmd/tendermint/commands/run_node.go]: I just changed runNodeCmd to cmd, so I could use this elsewhere.  Any other change was a typo.

File: cmd/tendermint/commands/run_node.go (hunk 3 above)
Comment (ethanfrey) [inline cmd/tendermint/commands/run_node.go]: yes, kill this wait loop!

Comment (ebuchman) [inline cmd/tendermint/commands/run_node.go]: since its technically breaking, ill wait until 0.11 https://github.com/tendermint/tendermint/issues/562

**Pull request merged**